File: 171-excel-sheet-column-number/171-excel-sheet-column-number.py
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        res = 0
        pow_26 = 1
        for i in range(len(columnTitle)-1, -1, -1):
            c = columnTitle[i]
            res += (ord(c) - ord("A") + 1) * pow_26
            pow_26 *= 26
        return res

File: 171-excel-sheet-column-number/171-excel-sheet-column-number.cpp
class Solution {
public:
    int titleToNumber(string columnTitle) {
        long long res = 0;
        long long pow_26 = 1;
        char c;
        for (int i=(columnTitle.length()-1); i>-1; i--) {
            c = columnTitle[i];
            res += ((c - 'A') + 1) * pow_26;
            pow_26 *= 26;
        }
        return res;
    }
};
